packages/components/src/internal/components/navigation/ProductMenu.tsx

```diff
@@ -13,100 +13,328 @@
+    const [menuModel, setMenuModel] = useState<ProductMenuModel>(new ProductMenuModel({ containerId: container.id }));
+    const contentRef = useRef<HTMLDivElement>();
+
+    useEffect(() => {
+        if (!menuModel.isLoaded) return;
+        let height = 400; // match navbar.scss product-menu-content min-height
+        const maxHeight = window.innerHeight * 0.8;
+        const sections = Array.from(contentRef.current.getElementsByClassName('menu-section'));
+        sections.forEach(section => (height = Math.max(height, section.clientHeight)));
+        contentRef.current.style.height = Math.min(height, maxHeight) + 'px';
+
+        // if the selected project is out of view, scrollIntoView
+        const activeProject = contentRef.current.getElementsByClassName('active')?.[0];
+        if (activeProject) {
+            if (activeProject.getBoundingClientRect().bottom > contentRef.current.getBoundingClientRect().bottom) {
+                contentRef.current.getElementsByClassName('active')?.[0].scrollIntoView({ behavior: 'smooth' });
+            }
+        }
+    }, [menuModel.isLoaded]);
+
+    useEffect(() => {
+        (async () => {
+            // no try/catch as the initMenuModel will catch errors and put them in the model isError/message
+            const menuModel_ = await api.navigation.initMenuModel(appProperties, moduleContext, container.id);
+            setMenuModel(menuModel_);
+        })();
+    }, [api.navigation, appProperties, container.id, moduleContext]);
+
+    const onFolderItemClick = useCallback(
+        async (folderItem: FolderMenuItem) => {
+            // return early if folderItem is already active
+            if (folderItem.id === menuModel.containerId) return;
+
+            setMenuModel(new ProductMenuModel({ containerId: folderItem.id })); // loading state, reset error
+
+            // no try/catch as the initMenuModel will catch errors and put them in the model isError/message
+            const containerPath = folderItem.id === container.id ? undefined : folderItem.path;
+            const menuModel_ = await api.navigation.initMenuModel(
+                appProperties,
+                moduleContext,
+                folderItem.id,
+                containerPath
+            );
+            setMenuModel(menuModel_);
+        },
+        [api.navigation, appProperties, container.id, menuModel.containerId, moduleContext]
+    );
+
+    const getSectionModel = useCallback(
+        (key: string): MenuSectionModel => menuModel.sections.find(section => section.key === key),
+        [menuModel]
+    );
+
+    const dashboardURL = useMemo(() => {
+        return showFolderMenu && appProperties.logoBadgeColorImageUrl;
+    }, [appProperties.logoBadgeColorImageUrl, showFolderMenu]);
+
+    return (
+        <div className={classNames('product-menu-content', className)} onClick={onClick} ref={contentRef}>
+            <div className="navbar-connector" />
+            {error && <Alert>{error}</Alert>}
```

Comment:
```suggestion
            {<Alert>{error}</Alert>}
```

Reply:
I agree that works. But I always find it easier for the jest test if I conditionally render the Alert. That way the test can check if the Alert exists or not instead of it always being there and needing to check something else.